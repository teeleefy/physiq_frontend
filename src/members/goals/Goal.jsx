import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
import '../styles/Goal.css'


function Goal({goal, memberId}){

    return(
        <>
            <ListGroupItem className="Goal-card">
            <Card className="Goal-card">
                <CardBody>
                    <CardText>
                        <p><b>Goal Name:</b> <NavLink className="Goal-name" to={`/member/${memberId}/goals/${goal.id}`} key={goal.id}>
                            {goal.goalName}
                        </NavLink></p>
                        <p><b>Details:</b> {goal.goalDetails}</p>
                    </CardText>
                </CardBody>
            </Card>
            </ListGroupItem>
        </>
     )
}

export default Goal;