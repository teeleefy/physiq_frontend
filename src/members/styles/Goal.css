.Goal-card {
    height: 15vh;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.Goal-label {
    font-size: 1.5rem;
}

.Goal-h1 {
    font-size: 3rem;
    font-weight: 600;
}

.Goals-navlink {
    text-decoration: none;
    color: white;
}