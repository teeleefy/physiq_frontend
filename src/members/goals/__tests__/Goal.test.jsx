import { renderWithRouter, screen } from '../../../../__tests__/test-tools';
import Goal from '../Goal'

let goal={goalName: "Make appointment with Dr. Jones", goalDetails: "Need to discuss medication and get labs done."}
let memberId=1000;

describe('Goal', () => {
  it('renders the Goal component', () => {
    renderWithRouter(<Goal goal={goal} memberId={memberId}/>)
    
    screen.debug(); // prints out the jsx in the Goal component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<Goal goal={goal} memberId={memberId}/>);
  expect(asFragment()).toMatchSnapshot();
});