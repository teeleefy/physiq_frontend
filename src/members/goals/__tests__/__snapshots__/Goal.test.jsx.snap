// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="Goal-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Goal Name:
          </b>
           
          <a
            class="Goal-name"
            href="/member/1000/goals/undefined"
          >
            Make appointment with Dr. Jones
          </a>
        </p>
        <p>
          <b>
            Details:
          </b>
           Need to discuss medication and get labs done.
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
